### PR TITLE
GH-43840: [CI] Add cuda group to tasks.yml and minor updates for new cuda runner image

### PR DESCRIPTION
**Pull request by raulcd**
### Rationale for this change

Trigger cuda jobs as a group on crossbow

### What changes are included in this PR?

Grouping of cuda tasks under `cuda` on tasks.yml.
We have also updated the cuda runner image used to https://github.com/voltrondata-labs/cuda-action-runner-builder/blob/bc1797368e02d98e4dc04de8afe41807e2171f3c/.github/workflows/cuda-dind-runners.yaml. It will run on Ubuntu 22.04, base Python is updated to 3.10, CUDA updated from 11.4.1 to 11.8.0

### Are these changes tested?

Via archery

### Are there any user-facing changes?

No
* GitHub Issue: #43840

### CODE REVIEW

**Comment by raulcd** [comment]
@github-actions crossbow submit -g cuda

**Comment by raulcd** [comment]
@github-actions crossbow submit test-cuda-python

**Comment by raulcd** [comment]
@github-actions crossbow submit test-cuda-python

**Comment by raulcd** [comment]
@github-actions crossbow submit test-cuda-python

**Comment by raulcd** [comment]
@github-actions crossbow submit test-cuda-python

**Comment by raulcd** [comment]
@github-actions crossbow submit test-cuda-python

**Comment by raulcd** [comment]
@github-actions crossbow submit test-cuda-python

**Comment by raulcd** [comment]
I am unsure why the gpu capabilities are not found:
```
DEBUG:archery:Executing `['docker', 'run', '--rm', '--gpus', 'all', '--shm-size', '2147483648', '-e', 'ARROW_BUILD_UTILITIES=OFF', '-e', 'ARROW_COMPUTE=ON', '-e', 'ARROW_CSV=ON', '-e', 'ARROW_CUDA=ON', '-e', 'ARROW_DATASET=ON', '-e', 'ARROW_ENABLE_TIMING_TESTS=OFF', '-e', 'ARROW_FILESYSTEM=ON', '-e', 'ARROW_GANDIVA=OFF', '-e', 'ARROW_GCS=OFF', '-e', 'ARROW_HDFS=ON', '-e', 'ARROW_JEMALLOC=ON', '-e', 'ARROW_JSON=ON', '-e', 'ARROW_ORC=OFF', '-e', 'ARROW_PARQUET=ON', '-e', 'ARROW_S3=OFF', '-e', 'ARROW_SUBSTRAIT=OFF', '-e', 'ARROW_WITH_OPENTELEMETRY=OFF', '-e', 'AWS_ACCESS_KEY_ID=***', '-e', 'AWS_SECRET_ACCESS_KEY=***', '-e', 'CCACHE_COMPILERCHECK=content', '-e', 'CCACHE_COMPRESS=1', '-e', 'CCACHE_COMPRESSLEVEL=6', '-e', 'CCACHE_DIR=/ccache', '-e', 'CCACHE_MAXSIZE=1G', '-e', 'GITHUB_ACTIONS=true', '-e', 'SCCACHE_BUCKET=***', '-e', 'SCCACHE_REGION=***', '-e', 'SCCACHE_S3_KEY_PREFIX=sccache', '-v', '/runner/_work/crossbow/crossbow/arrow:/arrow', '-v', 'ubuntu-ccache:/ccache', '-e', 'SETUPTOOLS_SCM_PRETEND_VERSION=18
docker: Error response from daemon: could not select device driver "" with capabilities: [[gpu]].
Error: docker run --rm --gpus all --shm-size 2147483648 -e ARROW_BUILD_UTILITIES=OFF -e ARROW_COMPUTE=ON -e ARROW_CSV=ON -e ARROW_CUDA=ON -e ARROW_DATASET=ON -e ARROW_ENABLE_TIMING_TESTS=OFF -e ARROW_FILESYSTEM=ON -e ARROW_GANDIVA=OFF -e ARROW_GCS=OFF -e ARROW_HDFS=ON -e ARROW_JEMALLOC=ON -e ARROW_JSON=ON -e ARROW_ORC=OFF -e ARROW_PARQUET=ON -e ARROW_S3=OFF -e ARROW_SUBSTRAIT=OFF -e ARROW_WITH_OPENTELEMETRY=OFF -e AWS_ACCESS_KEY_ID=*** -e AWS_SECRET_ACCESS_KEY=*** -e CCACHE_COMPILERCHECK=content -e CCACHE_COMPRESS=1 -e CCACHE_COMPRESSLEVEL=6 -e CCACHE_DIR=/ccache -e CCACHE_MAXSIZE=1G -e GITHUB_ACTIONS=true -e SCCACHE_BUCKET=*** -e SCCACHE_REGION=*** -e SCCACHE_S3_KEY_PREFIX=sccache -v /runner/_work/crossbow/crossbow/arrow:/arrow -v ubuntu-ccache:/ccache -e SETUPTOOLS_SCM_PRETEND_VERSION=18.0.0.dev221 apache/arrow-dev:amd64-ubuntu-22.04-cuda-11.8.0-python-3 /bin/bash -c   /arrow/ci/scripts/cpp_build.sh /arrow /build &&
  /arrow/ci/scripts/python_build.sh /arrow /build &&
  /arrow/ci/scripts/python_test.sh /arrow exited with non-zero exit code 125
  ```
@assignUser do you recall something like this in the past?

**Comment by raulcd** [comment]
we are going to revert the images to the old ones while @amaldonadomat and I investigate if there is any issue with the base runner images

**Comment by assignUser** [comment]
@raulcd the only issue I remember that gave me trouble was a mismatch of the host and docker cuda version. I don't remember the error message but do remember that it was weirdly hard to track down...

**Comment by kou** [comment]
@github-actions crossbow submit -g cuda

**Comment by raulcd** [comment]
@github-actions crossbow submit -g cuda

File: dev/tasks/tasks.yml [hunk 1]
```diff
@@ -1522,6 +1525,9 @@ tasks:
     ci: github
     template: docker-tests/github.cuda.yml
     params:
+      env:
+        UBUNTU: 22.04
+        CUDA: 11.8.0
```
Comment (pitrou) [inline dev/tasks/tasks.yml:1530]: Can we keep the same UBUNTU and CUDA version for both jobs?

Comment (raulcd) [inline dev/tasks/tasks.yml:1530]:
```suggestion
```

**Comment by raulcd** [comment]
@github-actions crossbow submit -g cuda

File: dev/tasks/tasks.yml (hunk 1 above)
Comment (raulcd) [reply dev/tasks/tasks.yml]: I've reverted back to defaults. This was just some testing we were doing last week when we tried to update the runners and it was failing

**Review by pitrou — APPROVED** [review]
+1, thanks for fixing this @raulcd

**Pull request merged**